mrp_subcontracting_skip_no_negative/models/stock_move.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _action_done(self, cancel_backorder=False):
+        subcontract_moves = self.filtered(lambda x: x.is_subcontract).with_context(
+            skip_negative_qty_check=True
+        )
+        res = super(StockMove, self - subcontract_moves)._action_done(
+            cancel_backorder=cancel_backorder
+        )
+        res += super(StockMove, subcontract_moves)._action_done(
+            cancel_backorder=cancel_backorder
+        )
+        return res
```

Comment:
```suggestion
        moves_with_no_check = self.filtered(lambda x: x.is_subcontract).with_context(
            skip_negative_qty_check=True
        )
        # For rather unlikely occassions where linked production is not in the right
        # state.
        for move in moves_with_no_check:
            production_move = self.search([("move_dest_ids", "=", move.id)])
            if production_move.state == "assigned":
                continue
            moves_with_no_check -= move
        res = super(StockMove, self - moves_with_no_check)._action_done(
            cancel_backorder=cancel_backorder
        )
        res += super(StockMove, moves_with_no_check)._action_done(
            cancel_backorder=cancel_backorder
        )
        return res
```

Reply:
@yostashiro  When I checked the behavior, the production_move is always in confirmed state for finished product and never skip the check for subcontracting receipt. After validated the subcontracting receipt, this production_move is changed to done state.